src/cuda/ptx.cu

```diff
@@ -0,0 +1,262 @@
+                                to_be_processed_size, cudaMemcpyHostToDevice);
+        if (cudaStatus != cudaSuccess) {
+            fprintf(stderr, "cudaMemcpy failed!");
+            break;
+        }
+        cudaStatus = cudaMemcpy(d_iv, (iv + (processed_piece_size / 512)),
+                                (to_be_processed_size / 128), cudaMemcpyHostToDevice);
+        if (cudaStatus != cudaSuccess) {
+            fprintf(stderr, "cudaMemcpy failed!");
+            break;
+        }
+        // the reason for the extra division by 4 is:
+        // we are doing pointer arithmetic here. Type of `piece` and `iv` are unsigned int, and unsigned int
+        // is allocating 4 bytes. So actually, iv+1 reaches to next unsigned int, which is 4 bytes later
+        // and we have computed the actual size. We have to divide our computations by 4 in here
+
+        sloth256_189_encode<<<block_count, thread_count>>>(d_piece, d_iv);  // calling the kernel
+
+        cudaStatus = cudaGetLastError();  // Check for any errors launching the kernel
+        if (cudaStatus != cudaSuccess) {
+            fprintf(stderr, "Kernel launch failed: %s\n", cudaGetErrorString(cudaStatus));
+            break;
+        }
+
+        cudaStatus = cudaDeviceSynchronize();
+        if (cudaStatus != cudaSuccess) {
+            fprintf(stderr, "cudaDeviceSynchronize returned error code %d, aborting...\n", cudaStatus);
+            break;
+        }
+
+        cudaStatus = cudaMemcpy((piece + (processed_piece_size / 4)), d_piece,
+                                to_be_processed_size, cudaMemcpyDeviceToHost);
+        // copy back the result to host, again extra division by 4 because of pointer arithmetic
+        if (cudaStatus != cudaSuccess) {
+            fprintf(stderr, "cudaMemcpy failed!");
+            break;
+        }
+
+        processed_piece_size += to_be_processed_size;  // update the processed_piece_size
+        remaining_piece_size -= to_be_processed_size;  // likewise :)
+
+        if (remaining_piece_size == 0) {  // successful!
+            break;  // Hurry! Get out of the loop
+        }
+    }
+
+    cudaFree(d_piece);  // clean-up
+    cudaFree(d_iv);  // clean-up
+
+    return cudaStatus;  // cudaStatus is 0 if there is no error, 1 if there is error
+}
+
+
+extern "C" bool test_1x1_cuda(unsigned int piece[], size_t len,
+                              const unsigned int iv[32], size_t layers)
+{
+    u256* d_piece;
+    u256* d_iv;
+
+    cudaMalloc(&d_piece, len);
+    cudaMalloc(&d_iv, 32);
+
+    cudaMemcpy(d_piece, piece, len, cudaMemcpyHostToDevice);
+    cudaMemcpy(d_iv, iv, 32, cudaMemcpyHostToDevice);
+
+    sloth256_189_encode<<<1, 1 >>>(d_piece, d_iv);  // calling the kernel
+
+    if (cudaDeviceSynchronize() == cudaSuccess)
+        cudaMemcpy(piece, d_piece, len, cudaMemcpyDeviceToHost);
+
+    cudaFree(d_piece);
+    cudaFree(d_iv);
+
+    return true;
+}
+
+
+
+
+#ifdef STANDALONE_DEMO
```

Comment:
I don't think we need standalone demo anymore, definitely not here.

Reply:
Alright, removing all the below code as well for standalone demo